Library/Homebrew/os/mac/version.rb

```diff
@@ -43,6 +43,10 @@ def pretty_name
 
       # For OS::Mac::Version compatibility
       def requires_nehalem_cpu?
+        unless Hardware::CPU.intel?
+          raise "Unexpected architecture: #{Hardware::CPU.arch}. This only works with Intel architecture."
+        end
```

Comment:
It's an interesting one this. I spent a few minutes debating with myself wheether this should instead return `false`.

In some cases raising makes sense as the formula might need some intention, in other cases returning `false` might actually make some formulae just work out-of-the-box.

Reply:
Yeah, I agree. My rationale was that I’ve found a couple of formulae on homebrew-core doing stuff like: https://github.com/Homebrew/homebrew-core/blob/9e6f3fd50144b07f4bda3fdfed77f594c207a59e/Formula/john-jumbo.rb#L46

(`requires_sse4?` is an alias to `requires_nehalem_cpu?`.)

All similar usages I’ve found on homebrew-core do something `unless requires_sse4?`, just like in the example. So I thought we’d rather want those to fail safely.